central/externalbackups/plugins/s3compatible/s3compatible_test.go

```diff
@@ -0,0 +1,45 @@
+package s3compatible
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEndpointValidation(t *testing.T) {
+	testCases := map[string]struct {
+		endpoint          string
+		sanitizedEndpoint string
+		shouldError       bool
+	}{
+		"with https prefix": {
+			endpoint:          "https://play.min.io",
+			sanitizedEndpoint: "https://play.min.io",
+		},
+		"with http prefix": {
+			endpoint:          "http://play.min.io",
+			sanitizedEndpoint: "http://play.min.io",
+		},
+		"without prefix": {
+			endpoint:          "play.min.io",
+			sanitizedEndpoint: "https://play.min.io",
+		},
```

Comment:
[nit] Maybe add a test case that tests the removal of trailing slash:
```suggestion
		"without prefix": {
			endpoint:          "play.min.io",
			sanitizedEndpoint: "https://play.min.io",
		},
		"with trailing slash": {
			endpoint:          "https://play.min.io/",
			sanitizedEndpoint: "https://play.min.io",
		},
```

Reply:
good idea, done